### PR TITLE
client/shell: Start instance if it's not running

### LINKED ISSUE #621: `multipass shell <instance_name>` should start the instance if it's not running
`shell` should start a `STOPPED` or `SUSPENDED` instance instead of reporting an error.

### PR DESCRIPTION
**Pull request by townsend2010**
Fixes #621

### CODE REVIEW

File: src/daemon/daemon.cpp [hunk 1]
```diff
@@ -1353,8 +1353,7 @@ try // clang-format on
         auto& vm = it->second;
         if (!mp::utils::is_running(vm->current_state()))
         {
-            return grpc::Status(grpc::StatusCode::FAILED_PRECONDITION,
-                                fmt::format("instance \"{}\" is not running", name), "");
+            return grpc::Status(grpc::StatusCode::ABORTED, fmt::format("'{}' needs starting", name));
```
Comment (gerboland) [inline src/daemon/daemon.cpp]: I'd prefer message said what is current state, rather than what might need to be done. I.e. "{} is not running"

Comment (gerboland) [reply src/daemon/daemon.cpp]: Hmm, this is printed to the console, as I see "shell failed: 'b' needs starting" before the spinner.
That's not great UI, as nothing has failed really. Can that message be suppressed?

Comment (townsend2010) [reply src/daemon/daemon.cpp]: @gerboland,

Sure, we can suppress the message.  I think the "Starting foo" message on the console would be enough to clue the user that something is happening and tells why the shell hasn't started.

**Comment by townsend2010** [comment]
@gerboland,

Regarding tests, yeah, we need to do a bunch of work in these types of places to enable testing, ie, mocking many things, etc.  It's something we really, really need to do, but this is not the place to do it.

**Comment by gerboland** [comment]
Re: tests, ok

File: src/daemon/daemon.cpp (hunk 1 above)
Comment (townsend2010) [reply src/daemon/daemon.cpp]: Ok, switched back to the original message **and** the client doesn't print this out when `grpc::StatusCode::ABORTED` is returned.

**Comment by townsend2010** [comment]
@gerboland, this is ready for another review pass.  Thanks!

**Pull request merged**